cvmfs/download.cc

```diff
@@ -177,25 +177,27 @@ static size_t CallbackCurlHeader(void *ptr, size_t size, size_t nmemb,
   //          header_line.c_str());
 
   // Check http status codes
-  if (HasPrefix(header_line, "HTTP/1.", false)) {
-    if (header_line.length() < 10)
+  if (HasPrefix(header_line, "HTTP/", false)) {
+    if (header_line.length() < 12) {
       return 0;
+    }
 
     unsigned i;
     for (i = 8; (i < header_line.length()) && (header_line[i] == ' '); ++i) {}
 
     // TODO(jblomer): consolidate the code
```

Comment:
This TODO can go, as the patch-set below "consolidate[s] the code"... It now checks the parsed HTTP status code rather than doing funny offset magic to check individual status code positions.